setup.py

```diff
@@ -19,8 +20,12 @@
     os.path.join('erfa', 'ufunc.c'),
 ]
 
+
 # build with Py_LIMITED_API unless explicitly disabled
-USE_PY_LIMITED_API = os.getenv("PYERFA_LIMITED_API", "1") != "0"
+USE_PY_LIMITED_API = (
+    (not sysconfig.get_config_var("Py_GIL_DISABLED")) and
```

Comment:
Thanks for the PR! The link you give makes sense, but I'm still confused - this piece is about building, not running, and might we not want to build without the limited API on a regular, GIL-enabled python? cc @neutrinoceros, since he just put this in.

Reply:
I don't know of a reason to build with the GIL and without Limited API, but the reason I specifically didn't implement this automatic switch is that it's only expected to be relevant for Python 3.13t, as 3.14t is expected to support the Limited API. _That said_, I made this choice before the guide had a recommendation there (or maybe before I learned about it), and maybe it's easier to follow a convention than having our own solution in place.

Reply:
My intent with this change was to preserve the ability to build with GIL and with not-limited.

As a random person trying to build packages (https://github.com/tacaswell/build_the_world), not having to set an pyerfa specific ENV is very convenient!

Reply:
OK, that makes sense - thanks, @neutrinoceros and @tacaswell! 

@neutrinoceros - do we still need your special variable?

Reply:
It's not used anywhere yet, and @tacaswell´s argument makes a lot of sense, so I say let's roll with it. Thanks !

Reply:
@neutrinoceros - So, remove your environment variable then in favor of this? (Right now, setting the environment variable is still kept as an option.)

Reply:
Yes !

Reply:
@tacaswell - are you able to make the small change requested? (Remove support for @neutrinoceros's environment variable.) If not, we can do it.